bin/generate-metadata.py

```diff
@@ -1,38 +1,49 @@
 #!/usr/bin/env python3
+from __future__ import annotations
+
 import grp
 import os
 import pwd
 import socket
+import sys
+import time
 
 Environ = dict[str, str]
 
 
 def relevant_env_var(var: str) -> bool:
     # prefixes and suffixes of environment vars that we may want during
     # debugging, test-result analysis, or traceback normalization
-    return var.endswith(("HOME", "PATH")) or var.startswith(
-        (
-            "USER",
-            "HOME",
-            "HOST",
-            "XDG",
-            "LOG",
-            "PWD",
-            "TMP",
-            "PYTHON",
-            "GITHUB",
-            "RUNNER",
-            "CI",
-            "MATRIX",
-            "TEST",
-        )
-    )
+    return var.endswith(("HOME", "PATH")) or var.startswith((
+        "USER",
+        "HOME",
+        "HOST",
+        "XDG",
+        "LOG",
+        "PWD",
+        "TMP",
+        "PYTHON",
+        "PYTEST",
+        "PIP",
+        "NODE",
+        "GITHUB",
+        "RUNNER",
+        "CI",
+        "MATRIX",
+        "TEST",
+    ))
 
 
 def get_env(env: Environ) -> Environ:
     """Keep the relevant parts of the environment variables."""
-    relevant_vars = sorted(var for var in env if relevant_env_var(var))
-    return {var: env[var] for var in relevant_vars}
+    result: Environ = {}
+
+    for var, val in sorted(env.items()):
+        if relevant_env_var(var):
+            result[var] = env[var]
+        else:
+            print(f"skipping irrelevant env var: {var}={val}", file=sys.stderr)
```

Comment:
What's the purpose of keeping the print around?

Reply:
It's a debug statement. These lines only show up if you expand the "action-collect-test-data" section of the log, which I'd only do if I want this kind of detail.

Reply:
Yes! `stdout` is reserved for data (as oposed to logging). In this case, it's later fed to a json parser. `stderr` was invented so that we had somewhere to send user-bound logs, apart from data.